### ISSUE TITLE
C#: "Find All References" Breaks In switch Statement

### ISSUE THREAD

**Issue by ericmutta** [dotnet/roslyn#54791]
**Version Used**:

Visual Studio 2019 Community v16.10.0

**Steps to Reproduce**:

Using the following code:

```csharp
namespace Test
{
  public static class Program
  {
    public enum Foo
    {
      Bar,
      Baz,
    }
    public static void Main() {
      var f = Foo.Bar;

      switch(f) {
        case Foo.Bar:
          break;
        case Foo.Baz:
          break;
      }
    }
  }
}
```

If you right-click on the member `Foo.Bar` within the enum definition here:

![image](https://user-images.githubusercontent.com/20465797/125462700-ba1f3a57-4932-49ed-88c4-19dba00bbc95.png)

...then choose "Find All References" the result window shows 3 results as expected and its title reads "Bar references":

![image](https://user-images.githubusercontent.com/20465797/125463115-20e3b085-4c30-426b-aa04-d06c77d21d0c.png)

The same expected result is produced when you do "Find All References" for `Foo.Bar` on the line where it is assigned to a variable:

![image](https://user-images.githubusercontent.com/20465797/125463665-87f68327-67ba-4ecb-9ac5-f1e0740085ae.png)

If however, you do "Find All References" using the `Foo.Bar` instance in the `case` branch of the `switch` statement here:

![image](https://user-images.githubusercontent.com/20465797/125464070-461b70a2-d21b-477f-87b9-db7a9a4c5313.png)

...the result window title reads "case Foo.Bar references" (not "Bar references" like in the other scenarios) and only shows one result:

![image](https://user-images.githubusercontent.com/20465797/125464360-10948556-df85-4375-a032-4bfd5c89b7ec.png)

**Expected Behavior**:

For as long as I can remember, "Find All References" has always returned ALL references to an identifier without regard to the context its used in and I should be able to invoke it and get the same results from all locations where an identifier is used.

**Actual Behavior**:

As shown above, "Find All References" for an enum member (I haven't tested other kinds of identifiers) that's used in a `case` branch of a `switch` statement doesn't return all references (in real code this almost led me to delete an enum member under the assumption it wasn't used elsewhere!).

**Comment by Youssef1313** [comment]
This was already fixed in v16.11. https://github.com/dotnet/roslyn/issues/54407, https://github.com/dotnet/roslyn/pull/53338.

**Comment by ericmutta** [comment]
Many thanks @Youssef1313 for the quick follow-up. Glad to hear others found the issue and it has been fixed, I will close this issue to avoid them piling up :+1:
